### ISSUE TITLE
KeyboardMarkup's SerializationException in 5.1 & 5.2

### ISSUE THREAD

**Issue by tonycode** [omarmiatello/telegram#2]
## 🐛 Describe the bug

I have a pet-project well working on your great telegramBot library 🔥 version **4.8** 👌

While trying to update it to the latest **5.2** version I've came across to a very strange bug - json **SerializationException** when `sendMessage` with any keyboard markup.

I've simplified the sources to the minimum and the bug persists:

## ⚠️ Current behavior

```
Exception in thread "main" kotlinx.serialization.SerializationException: Serializer for class 'KeyboardOption' is not found.
Mark the class as @Serializable or provide the serializer explicitly.
        at kotlinx.serialization.internal.Platform_commonKt.serializerNotRegistered(Platform.common.kt:91)
        at kotlinx.serialization.ContextualSerializer.serializer(ContextualSerializer.kt:51)
        at kotlinx.serialization.ContextualSerializer.serialize(ContextualSerializer.kt:61)
        at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeSerializableValue(StreamingJsonEncoder.kt:211)
        at kotlinx.serialization.encoding.Encoder$DefaultImpls.encodeNullableSerializableValue(Encoding.kt:302)
        at kotlinx.serialization.encoding.AbstractEncoder.encodeNullableSerializableValue(AbstractEncoder.kt:18)
        at kotlinx.serialization.encoding.AbstractEncoder.encodeNullableSerializableElement(AbstractEncoder.kt:90)
        at com.github.omarmiatello.telegram.TelegramRequest$SendMessageRequest$$serializer.serialize(TelegramModels.kt:3279)
        at com.github.omarmiatello.telegram.TelegramRequest$SendMessageRequest$$serializer.serialize(TelegramModels.kt:3279)
        at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeSerializableValue(StreamingJsonEncoder.kt:211)
        at kotlinx.serialization.json.Json.encodeToString(Json.kt:80)
        at com.github.omarmiatello.telegram.TelegramRequest$SendMessageRequest.toJsonForRequest(TelegramModels.kt:3291)
        at com.github.omarmiatello.telegram.TelegramClient.sendMessage(TelegramClient.kt:174)
        at com.github.omarmiatello.telegram.TelegramClient.sendMessage$default(TelegramClient.kt:152)
        at com.demo.MainKt$main$1.invokeSuspend(main.kt:23)
```

## ✅ Expected behavior

- the message with keyboardMarkup sent successfully

## 💣 Steps to reproduce

**main.kt**:

```Kotlin
package com.demo

import com.github.omarmiatello.telegram.InlineKeyboardButton
import com.github.omarmiatello.telegram.InlineKeyboardMarkup
import com.github.omarmiatello.telegram.TelegramClient
import kotlinx.coroutines.runBlocking


private const val TGBOT_TOKEN = "1234567890:ABCDEFGhijklmnopqrstuvwxyz123456"

fun main(args: Array<String>) {
    println("Hello World!")

    val tgClient = TelegramClient(TGBOT_TOKEN)

    runBlocking {
        val response = tgClient.getUpdates()
        if (response.ok) {
            val updates = response.result
            for (update in updates) {

                update.message?.let { message ->
                    tgClient.sendMessage(
                        chat_id = message.chat.id.toString(),
                        text = "I've got it!",
                        reply_to_message_id = message.message_id,
                        reply_markup = InlineKeyboardMarkup(listOf(
                            listOf(
                                InlineKeyboardButton("Button1", url = "https://google.com")
                            )
                        ))

                    )
                }

            }
        }
    }
}
```

**build.gradle.kts**:

```Kotlin
plugins {
    kotlin("jvm") version "1.5.20"
    application
}


group = "com.demo"
version = "1.0.0-SNAPSHOT"

application {
    mainClass.set("com.demo.MainKt")
}


repositories {
    mavenCentral()
    maven("https://jitpack.io")
}

dependencies {
    implementation("io.ktor:ktor-client-core:1.5.1")
    implementation("io.ktor:ktor-client-okhttp:1.5.1")

    implementation("com.github.omarmiatello.telegram:dataclass:5.2")
    implementation("com.github.omarmiatello.telegram:client:5.2")
}
```

gradle wrapper: `6.8-bin` & `7.1.1-all`

## 📱 Tech info

 - Library/App version: **5.1** & **5.2** are affected
 

- I've found some "nearly the same" Kotlin serialization issue with sealed classes - https://github.com/Kotlin/kotlinx.serialization/issues/1116 - but the last relates to Multiplatform/JS, not JVM.

- The root cause is with this parameter - `val reply_markup: @ContextualSerialization KeyboardOption? = null` - the Json serializer don't use contextual serialization, but trying to serialize superclass `KeyboardOption` (without `serialize()` as it is only a super-class without it's own fields) instead of `InlineKeyboardMarkup` (with kotlinx-serialization generated `serialize()` methods)

P.S. Adding a lot of related dependencies to the minimal demo don't help:

```
plugins {
    ...
    kotlin("plugin.serialization") version "1.5.20"
}

dependencies {
    implementation(platform("org.jetbrains.kotlin:kotlin-bom"))  // Align versions of all Kotlin components
    implementation("org.jetbrains.kotlin:kotlin-stdlib-jdk8")  // Use the Kotlin JDK 8 standard library
    implementation("org.jetbrains.kotlinx:kotlinx-serialization-json:1.2.0")
...
    implementation("io.ktor:ktor-client-json:1.5.1")
    implementation("io.ktor:ktor-serialization:1.5.1")
}
```

Curious how to solve this problem... Maybe it's required to turn on some `OptIn`/some serialization settings in the minimum demo code.

**Comment by omarmiatello** [comment]
I'm working on that

**Comment by omarmiatello** [comment]
It should be fixed in the next release `5.2.1`! - available in a few hours

**Comment by tonycode** [comment]
I have tested it (different ways of Keyboards usage) - works like a SwissWatch! A great KotlinMultiPlatform library 🔥
